.gitignore

```diff
@@ -39,6 +39,14 @@ sites/*/private
 # Ignore SimpleTest multi-site environment.
 sites/simpletest
 
+# Ignore default settings that are updated upon Drupal core updates.
+web/sites/default/default.settings.php
+web/sites/example.sites.php
+
+# Ignore generated files
```

Comment:
I am not sure it is worth ignoring `default.settings.php` because it seems to make it harder to switch branches when there are updates. I have had to change the permissions and deal with a conflict going between branches locally that have settings updates. (Feel free to tell me if there is a better way to manage that.)